src/AppInstallerRepositoryCore/Rest/RestSourceFactory.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "RestSourceFactory.h"
+#include "RestClient.h"
+#include "RestSource.h"
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Rest
+{
+	namespace
+	{
+		// The base class for data that comes from a rest based source.
+		struct RestSourceFactoryBase : public ISourceFactory
+		{
+			std::shared_ptr<ISource> Create(const SourceDetails& details, IProgressCallback& progress) override final
+			{
+				THROW_HR_IF(E_INVALIDARG, details.Type != RestSourceFactory::Type());
+				return CreateInternal(details, progress);
+			}
+
+			virtual std::shared_ptr<ISource> CreateInternal(const SourceDetails& details, IProgressCallback& progress) = 0;
+
+			void Add(SourceDetails& details, IProgressCallback& progress) override final
+			{
+				UNREFERENCED_PARAMETER(progress);
+
+				if (details.Type.empty())
+				{
+					// With more than one source implementation, we will probably need to probe first
+					details.Type = RestSourceFactory::Type();
+					AICLI_LOG(Repo, Info, << "Initializing source type: " << details.Name << " => " << details.Type);
+				}
+				else
+				{
+					THROW_HR_IF(E_INVALIDARG, details.Type != RestSourceFactory::Type());
+				}
+
+				// Check if URL is remote and secure
+				THROW_HR_IF(APPINSTALLER_CLI_ERROR_SOURCE_NOT_SECURE, Utility::IsUrlRemote(details.Arg) && !Utility::IsUrlSecure(details.Arg));

```

Comment:
Should probably require "remote" and secure in this instance.  The other source allows for local files to be served for the index and manifests.  Unless you plan to support that here, it should just be two lines, with the first requiring remote and the second secure (different errors).